### PR TITLE
Migrate product browse

### PR DESCRIPTION
**Pull request by kul3r4**

### CODE REVIEW

File: templates/css/product.css [hunk 1]
```diff
@@ -30,9 +43,9 @@ amp-carousel {
   left:36px;
   font-size:18px;
   font-weight:400;
-  color:#fff;
+  color:var(--color-text-light);
   text-transform:uppercase;
-  margin:0 16px;
+  margin:0 var(--space-2);
   height:32px;
```
Comment (sebastianbenz) [inline templates/css/product.css]: use `space-4`

File: templates/css/product.css [hunk 2]
```diff
@@ -49,8 +62,8 @@ amp-carousel {
 .header input[type=submit] {
   position:absolute;
   top:10px;
```
Comment (sebastianbenz) [inline templates/css/product.css]: Would it look OK to change this to space-1 or space-2?

File: templates/css/product.css [hunk 3]
```diff
@@ -65,8 +78,8 @@ amp-carousel {
...
```
Comment (sebastianbenz) [inline templates/css/product.css]: Would it look OK to change this to space-1 or space-2?

File: templates/css/product.css [hunk 4]
```diff
@@ -65,8 +78,8 @@ amp-carousel {
...
   right:24px;
```
Comment (sebastianbenz) [inline templates/css/product.css]: use space-3

File: templates/css/product.css [hunk 5]
```diff
@@ -65,8 +78,8 @@ amp-carousel {
...
-  background:#ededed;
-  padding:0 8px;
+  background:var(--color-bg-light);
+  padding:0 var(--space-1);
   width:20vw;
   height:32px;
```
Comment (sebastianbenz) [inline templates/css/product.css]: use space-4

File: src/60_Samples_%26_Templates/Product_Browse_Page.html [hunk 6]
```diff
@@ -90,15 +98,15 @@
              display: block;
              height: 100px;
              padding: 0;
-             padding-right: 8px;
+             padding-right: var(--space-1);
            }
            .products amp-img {
              line-height: 24px;
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Product_Browse_Page.html]: use space-3

File: src/60_Samples_%26_Templates/Product_Browse_Page.html [hunk 7]
```diff
@@ -90,15 +98,15 @@
...
              font-weight: 400;
              font-size: 24px;
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Product_Browse_Page.html]: use space-3

File: src/60_Samples_%26_Templates/Product_Browse_Page.html [hunk 8]
```diff
@@ -127,7 +135,7 @@
          .autosuggest-box {
             position: absolute;
             width: 100%;
-            background-color: #fafafa;
+            background-color: var(--color-text-light);
             box-shadow: 0px 2px 6px rgba(0,0,0,.3);
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Product_Browse_Page.html]: use standard box-shadow or does it need to look different?

File: src/60_Samples_%26_Templates/Product_Browse_Page.html [hunk 9]
```diff
@@ -127,7 +135,7 @@
...
           }
           .hidden {
```
Comment (sebastianbenz) [inline src/60_Samples_%26_Templates/Product_Browse_Page.html]: Do we still need the hidden rule? Or could we replace this with the attribute?

File: templates/css/product.css [hunk 10]
```diff
@@ -19,7 +32,7 @@ amp-carousel {
   left:0;
   font-size:18px;
   font-weight:700;
-  color:#fff;
+  color: var(--color-text-light);
   text-transform:uppercase;
   padding:13px
```
Comment (sebastianbenz) [inline templates/css/product.css]: Would it make sense to replace this either with space-2 or space-3?

**Comment by kul3r4** [comment]
@sebastianbenz your comment regarding " Can you also migrate the colors.": which ones you mean? I added the rule in product.css and then used them in the code

File: templates/css/product.css (hunk 1 above)
Comment (kul3r4) [reply templates/css/product.css]: done

File: templates/css/product.css (hunk 2 above)
Comment (kul3r4) [reply templates/css/product.css]: done

File: templates/css/product.css (hunk 3 above)
Comment (kul3r4) [reply templates/css/product.css]: done

File: templates/css/product.css (hunk 4 above)
Comment (kul3r4) [reply templates/css/product.css]: done

File: templates/css/product.css (hunk 5 above)
Comment (kul3r4) [reply templates/css/product.css]: done

File: src/60_Samples_%26_Templates/Product_Browse_Page.html (hunk 6 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Product_Browse_Page.html]: done

File: src/60_Samples_%26_Templates/Product_Browse_Page.html (hunk 7 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Product_Browse_Page.html]: done

File: src/60_Samples_%26_Templates/Product_Browse_Page.html (hunk 9 above)
Comment (kul3r4) [reply src/60_Samples_%26_Templates/Product_Browse_Page.html]: removed

**Comment by sebastianbenz** [comment]
Demo is still broken. Am I doing something wrong?

Background is no longer transparent:

screenshot 2019-01-08 at 13 54 31

This seems wrong here:

screenshot 2019-01-08 at 13 54 26

Alignment off: 

screenshot 2019-01-08 at 13 54 23

Wrong colors:

screenshot 2019-01-08 at 13 54 18

Inconsistent styling:

screenshot 2019-01-08 at 13 54 14

Error in the console:

screenshot 2019-01-08 at 13 54 08

Animation off:

screenshot 2019-01-08 at 13 56 41

Inconsistent alignment and too much whitespace below and left/right.

screenshot 2019-01-08 at 13 57 10

**Comment by kul3r4** [comment]
@sebastianbenz for the wrong colors and inconsistent styling I don't have a particular css causing that but I believe it's amp-start, maybe because we import it in the <style amp-boilerplate>?

**Comment by kul3r4** [comment]
@sebastianbenz fixed the layout. The css now has become too bigger, not sure how to fix it.

**Comment by sebastianbenz** [comment]
You can ignore the CSS size limit. That hopefully is not going to be a problem with amp.dev as it's no longer using ampstart.

**Comment by kul3r4** [comment]
@sebastianbenz Applied all the suggestions

File: templates/css/product.css [hunk 11]
```diff
@@ -48,45 +61,45 @@ amp-carousel {
...
   border-radius:4px 0
 }
 
 .header input[type=search],.header input[type=text] {
   position:absolute;
-  top:10px;
-  right:24px;
-  background:#ededed;
-  padding:0 8px;
+  top:var(--space-1);
```
Comment (sebastianbenz) [inline templates/css/product.css]: I was wrong, this needs to be 10px for the input field to be centered vertically

**Comment by sebastianbenz** [comment]
And one more thing: can you use the same layout for the product list as in the original version (padding & margin).

**Comment by kul3r4** [comment]
Since you suggested to make the header with color-primary, which color should I use for the button?

**Comment by sebastianbenz** [comment]
The build fails because the product sample has gotten too big, which needs to skip the validation.

**Comment by sebastianbenz** [comment]
Does color-secondary look alright for the button? Or maybe vice versa?

**Comment by kul3r4** [comment]
@sebastianbenz it would look like this 
header-light-blue

**Comment by sebastianbenz** [comment]
Looks good! Can you post a full screenshot as I can't reproduce it locally.

**Comment by kul3r4** [comment]
@sebastianbenz it looks like:
product-browse-page

I also pushed the latest changes

**Comment by sebastianbenz** [comment]
Looks good. Two things:

1. Can you add the same margins as in the original version (around
headings, sections, ...)
2. Can you align the menu icon and product.

>

**Comment by kul3r4** [comment]
@sebastianbenz I missed those, sorry!
here is the new version
product-browse-page-2

**Comment by sebastianbenz** [comment]
Perfect. thanks!

**Pull request merged**